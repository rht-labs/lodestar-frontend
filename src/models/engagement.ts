export interface Engagement {
  name: string;
}
export class Engagement {
  constructor({ name }: Engagement) {
    this.name = name;
  }
  name: string;
}
