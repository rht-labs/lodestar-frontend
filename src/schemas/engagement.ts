import faker from 'faker/locale/en_US';
import { LaunchData } from './launch_data';
import { GitCommit } from './git_commit';
import { CreationDetails } from './creation_details';
import { ClusterStatus } from './cluster_status';
export interface Engagement {
  additional_details?: string;
  archive_date: Date;
  commits: GitCommit[];
  customer_contact_email: string;
  customer_contact_name: string;
  customer_name: string;
  description: string;
  end_date: Date;
  engagement_lead_email: string;
  engagement_lead_name: string;
  engagement_users: Array<any>;
  last_update: string;
  location: string;
  mongo_id?: string;
  ocp_cloud_provider_name: string;
  ocp_cloud_provider_region: string;
  ocp_cluster_size: string;
  ocp_persistent_storage_size: string;
  ocp_sub_domain: string;
  ocp_version: string;
  project_id: number;
  project_name: string;
  start_date: Date;
  technical_lead_email: string;
  technical_lead_name: string;
  launch?: LaunchData;
  creation_details: CreationDetails;
  last_update_by_name: string;
  suggested_subdomain?: string;
  status: ClusterStatus;
}
export abstract class Engagement {
  static fromFake(staticData = false): Engagement {
    return {
      additional_details: staticData
        ? 'Additional information here'
        : faker.lorem.paragraphs(2),
      archive_date: staticData ? new Date(2020, 6, 30) : faker.date.recent(),
      commits: [],
      customer_contact_email: staticData
        ? 'bob@doe.com'
        : faker.internet.email(),
      customer_contact_name: staticData
        ? 'Bob Doe'
        : `${faker.name.firstName()} ${faker.name.lastName()}`,
      customer_name: staticData ? 'NASA' : faker.company.companyName(),
      description: staticData
        ? "It's rocket science"
        : faker.lorem.paragraphs(5),
      end_date: staticData ? new Date(2020, 6, 1) : faker.date.future(),
      engagement_users: staticData ? [] : [],
      engagement_lead_email: staticData
        ? 'alice@doe.com'
        : faker.internet.email(),
      engagement_lead_name: staticData
        ? 'Alice Doe'
        : `${faker.name.firstName()} ${faker.name.lastName()}`,
      last_update: staticData ? '2020-01-01' : new Date().toString(),
      location: staticData
        ? 'Nashville, TN'
        : `${faker.address.city()}, ${faker.address.stateAbbr()}`,
      mongo_id: '1',
      ocp_cloud_provider_name: staticData ? 'AWS' : 'AWS',
      ocp_cloud_provider_region: staticData
        ? 'N. Virginia'
        : faker.lorem.word(),
      ocp_cluster_size: staticData ? 'Large' : 'Large',
      ocp_persistent_storage_size: staticData
        ? '100G'
        : `${faker.random.number()}G`,
      ocp_sub_domain: staticData ? 'ordoabchao' : faker.lorem.word(),
      ocp_version: staticData ? '4.4' : faker.random.number().toString(),
      project_id: staticData ? 1 : faker.random.number(),
      project_name: staticData ? 'Boots on the Moon' : faker.company.bsNoun(),
      start_date: staticData ? new Date(2020, 5, 1) : faker.date.recent(),
      technical_lead_email: staticData ? 'eve@doe.com' : faker.internet.email(),
      technical_lead_name: staticData
        ? 'Eve Doe'
        : `${faker.name.firstName()} ${faker.name.lastName()}`,
      creation_details: staticData
        ? {
            created_by_email: 'dwasinger@redhat.com',
            created_by_user: 'dwasinge',
            created_on: new Date(2020, 1, 1),
          }
        : {
            created_by_email: faker.internet.email(),
            created_by_user: `${faker.name.firstName()} ${faker.name.lastName()}`,
            created_on: faker.date.recent(),
          },
      last_update_by_name: staticData
        ? 'James Doe'
        : `${faker.name.firstName()} ${faker.name.lastName()}`,
      launch: staticData
        ? null
        : faker.random.boolean()
        ? {
            launched_by: faker.name.firstName(),
            launched_date_time: faker.date.recent(),
          }
        : null,
      status: ClusterStatus.fromFake(staticData),
    };
  }
}

export enum EngagementStatus {
  active = 'active',
  past = 'past',
  upcoming = 'upcoming',
}

export const getEngagementStatus = (
  engagement: Engagement
): EngagementStatus => {
  if (!engagement) {
    return null;
  }
  const Today = new Date();
  const { launch, end_date } = engagement;
  const hasLaunched = !!launch?.launched_date_time;
  if (hasLaunched && end_date >= Today) {
    return EngagementStatus.active;
  } else if (hasLaunched && end_date < Today) {
    return EngagementStatus.past;
  } else {
    return EngagementStatus.upcoming;
  }
};
