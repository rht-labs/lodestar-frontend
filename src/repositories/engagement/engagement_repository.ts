import { Engagement } from '../../models/engagement';

export abstract class EngagementRepository {
  abstract async fetchEngagements(): Promise<Engagement[]>;
}
