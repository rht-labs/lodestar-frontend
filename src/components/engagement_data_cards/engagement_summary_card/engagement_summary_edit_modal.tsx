import React from 'react';
import { Engagement } from '../../../schemas/engagement_schema';
import {
  Modal,
  ModalVariant,
  Button,
  TextInput,
  InputGroupText,
  InputGroup,
  FormGroup,
  Form,
  TextArea,
} from '@patternfly/react-core';
import { useModalVisibility } from '../../../context/edit_modal_visibility_context/edit_modal_visibility_hook';
import { EditModalTemplate } from '../../../layout/edit_modal_template';
import { CalendarAltIcon } from '@patternfly/react-icons';
import { useFeatures } from '../../../context/feature_toggles/feature_hook';
import { parse as parseDate, format as formatDate } from 'date-fns';
import { APP_FEATURES } from '../../../common/app_features';
import { useEngagements } from '../../../context/engagement_context/engagement_hook';
export interface EngagementSummaryEditModalProps {
  onChange: (fieldName: string, value: any) => void;
  formOptions: object;
  engagement: Engagement;
  isOpen: boolean;
}
export function EngagementSummaryEditModal(
  props: EngagementSummaryEditModalProps
) {
  const { requestClose } = useModalVisibility();
  const { hasFeature } = useFeatures();
  const getFormattedDate = (inputDate: Date | string = '') => {
    // Dates must be formatted YYYY-MM-DD for patternfly date picker.
    // They are coming back inconsistently from the backend,
    // so this function checks to see if the date needs to be formatted,
    // then formats the date appropriately
    if (!inputDate) {
      return;
    }
    if (inputDate instanceof Date) {
      return formatDate(inputDate, 'yyyy-MM-dd');
    } else if (inputDate.indexOf('-') > -1) {
      return inputDate;
    } else {
      return parseDate(inputDate, 'yyyyMMdd', new Date());
    }
  };
  const { saveEngagement, engagementFormState } = useEngagements();

  const onSave = () => {
    saveEngagement(engagementFormState);
    requestClose();
  };
  return (
    <Modal
      variant={ModalVariant.small}
      isOpen={props.isOpen}
      onClose={requestClose}
    >
      <EditModalTemplate
        actions={
          <div>
            <Button onClick={onSave}>Save</Button>
          </div>
        }
        title="Edit Engagement"
      >
        <Form>
          <FormGroup label="Description" fieldId="description">
            <TextArea
              disabled={!hasFeature(APP_FEATURES.writer)}
              name="description"
              id="description"
              aria-label="engagement description"
              placeholder="Description and notes for the Engagement"
              value={props.engagement.description || ''}
              onChange={e => props.onChange('description', e)}
            />
          </FormGroup>
          <FormGroup
            label="Location"
            fieldId="engagement-location"
            helperText="Where will this be held?"
          >
            <TextInput
              isDisabled={!hasFeature(APP_FEATURES.writer)}
              type="text"
              id="location"
              name="location"
              placeholder="e.g. Pasadena, CA"
              value={props.engagement.location || ''}
              onChange={e => props.onChange('location', e)}
            />
          </FormGroup>
          <FormGroup
            label="Start and End Dates"
            fieldId="engagement-dates"
            helperText="What is the duration?"
            isRequired
          >
            <InputGroup label="Engagement Duration">
              <InputGroupText component="label" htmlFor="engagement-duration">
                <CalendarAltIcon />
              </InputGroupText>
              <TextInput
                isDisabled={
                  !hasFeature(APP_FEATURES.writer) ||
                  !!(props.engagement as Engagement).launch
                }
                name="start_date"
                id="start_date"
                type="date"
                aria-label="The start date."
                value={getFormattedDate(props.engagement?.start_date) || ''}
                onChange={e =>
                  props.onChange('start_date', parseDate(e, 'yyyy-MM-dd', 0))
                }
              />
              <TextInput
                isDisabled={!hasFeature(APP_FEATURES.writer)}
                name="end_date"
                id="end_date"
                type="date"
                aria-label="The end date"
                value={getFormattedDate(props.engagement?.end_date) || ''}
                onChange={e =>
                  props.onChange('end_date', parseDate(e, 'yyyy-MM-dd', 0))
                }
              />
            </InputGroup>
          </FormGroup>
        </Form>
      </EditModalTemplate>
    </Modal>
  );
}
