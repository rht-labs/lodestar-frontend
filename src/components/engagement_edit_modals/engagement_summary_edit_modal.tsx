import React from 'react';
import { Engagement } from '../../schemas/engagement';
import { Modal, ModalVariant, Button, Form } from '@patternfly/react-core';
import { useModalVisibility } from '../../context/edit_modal_visibility_context/edit_modal_visibility_hook';
import { EditModalTemplate } from '../../layout/edit_modal_template';
import { DescriptionFormField } from '../engagement_form_fields/description';
import { LocationFormField } from '../engagement_form_fields/location';
import { EngagementStartEndDateFormField } from '../engagement_form_fields/engagement_start_end_date';
import { EngagementFormConfig } from '../../schemas/engagement_config';

export interface EngagementSummaryEditModalProps {
  onChange: (fieldName: string, value: any) => void;
  formOptions: EngagementFormConfig;
  engagement: Engagement;
  isOpen: boolean;
  onSave: (engagement: Engagement) => void;
}
export function EngagementSummaryEditModal(
  props: EngagementSummaryEditModalProps
) {
  const { requestClose } = useModalVisibility();

  const onSave = () => {
    props.onSave(props.engagement);
    requestClose();
  };
  return (
    <Modal
      variant={ModalVariant.small}
      isOpen={props.isOpen}
      onClose={requestClose}
      title="Engagement Summary"
    >
      <EditModalTemplate
        actions={
          <div>
            <Button onClick={onSave}
                    data-cy={"save_summary_card"}>
              Save
            </Button>
          </div>
        }
      >
        <Form>
          <DescriptionFormField
            onChange={props.onChange}
            engagement={props.engagement}
          />
          <LocationFormField
            onChange={props.onChange}
            engagement={props.engagement}
          />
          <EngagementStartEndDateFormField
            onChange={props.onChange}
            formOptions={props.formOptions}
            engagement={props.engagement}
          />
        </Form>
      </EditModalTemplate>
    </Modal>
  );
}
