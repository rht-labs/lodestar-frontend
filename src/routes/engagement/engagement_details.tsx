import React, { useEffect } from 'react';
import { Engagement } from '../../schemas/engagement';
import { useEngagement } from '../../context/engagement_context/engagement_hook';
import { Route, useParams, Switch, useRouteMatch } from 'react-router';
import { getValidatorsFromEngagementFormConfig } from '../../common/config_validator_adapter';
import {
  ValidationProvider,
  FormValidator,
} from '../../context/validation_context/validation_context';
import { EngagementDetailsViewTemplate } from '../../layout/engagement_details_view';
import { EngagementOverview } from './overview';
import { EngagementJsonDump } from './json_dump';
import { EngagementJsonSerializer } from '../../serializers/engagement/engagement_json_serializer';

export interface EngagementViewProps {
  currentEngagement?: Engagement;
}

export interface EngagementDetailViewProps {
  error: Error;
  currentEngagement: Engagement;
  createEngagementPoll: (
    engagement: Engagement
  ) => Promise<{ cancel: () => void }>;
  getEngagement: (
    customer_name: string,
    project_name: string
  ) => Promise<Engagement>;
}
export const validatorsWithDateValidators = (
  validators: FormValidator,
  dates: Pick<Engagement, 'start_date' | 'end_date'>
): FormValidator => {
  return {
    ...validators,
    end_date: [
      ...(validators['end_date'] ?? []),
      endDate => {
        const result =
          dates?.start_date > endDate
            ? 'End date must be after the start date'
            : null;
        return result;
      },
    ],
    archive_date: [
      ...(validators['archive_date'] ?? []),
      archiveDate => {
        const result =
          archiveDate < dates?.end_date
            ? 'Archive date must be after the end date'
            : null;
        return result;
      },
    ],
  };
};

export const EngagementDetailView = () => {
  const { project_name, customer_name } = useParams<{
    customer_name: string;
    project_name: string;
  }>();

  const {
    createEngagementPoll,
    saveEngagement,
    currentEngagement,
    currentChanges,
    engagementFormConfig,
  } = useEngagement({ projectName: project_name, customerName: customer_name });

  useEffect(() => {
    let engagementPoll;
    if (currentEngagement?.project_name && currentEngagement?.customer_name) {
      engagementPoll = createEngagementPoll(currentEngagement);
    }
    return async () => {
      (await engagementPoll)?.cancel?.();
    };
  }, [currentEngagement, createEngagementPoll]);

  const validators = getValidatorsFromEngagementFormConfig(
    engagementFormConfig
  );
  const { url } = useRouteMatch();
  const serializer = new EngagementJsonSerializer();
  const {
    commits,
    uuid,
    project_id,
    mongo_id,
    creation_details,
    status,
    launch,
    ...editableFields
  } = currentEngagement ?? {};
  const { end_date, start_date } = currentChanges;

  return (
    <ValidationProvider
      validators={validatorsWithDateValidators(validators, {
        end_date,
        start_date,
      })}
    >
      <EngagementDetailsViewTemplate
        engagement={currentEngagement}
        onSave={saveEngagement}
      >
        <Switch>
          <Route path={`${url}/json`}>
            <EngagementJsonDump
              json={JSON.stringify(
                serializer.serialize((editableFields ?? {}) as Engagement),
                null,
                2
              )}
              onSave={value => {
                saveEngagement(serializer.deserialize(JSON.parse(value)));
              }}
            />
          </Route>
          <Route path={`${url}/`}>
            <EngagementOverview />
          </Route>
        </Switch>
      </EngagementDetailsViewTemplate>
    </ValidationProvider>
  );
};
