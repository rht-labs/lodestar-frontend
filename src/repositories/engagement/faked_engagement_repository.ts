import { EngagementRepository } from './engagement_repository';
import { Engagement } from '../../models/engagement';
import faker from 'faker';

export class FakedEngagementRepository extends EngagementRepository {
  async fetchEngagements(): Promise<Engagement[]> {
    return new Array(8)
      .fill(null)
      .map(() => new Engagement({ name: faker.company.companyName() }));
  }
}
