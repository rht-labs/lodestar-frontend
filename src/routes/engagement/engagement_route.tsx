import React from 'react';
import { useAnalytics } from '../../context/analytics_context/analytics_context';
import { useSession } from '../../context/auth_context/auth_context';
import { useEngagementFormConfig } from '../../context/engagement_config_context/engagement_config_hook';
import { EngagementProvider } from '../../context/engagement_context/engagement_context';
import { useFeedback } from '../../context/feedback_context/feedback_context';
import { useServiceProviders } from '../../context/service_provider_context/service_provider_context';
import ScrollToTop from '../scroll_to_top';
import { EngagementDetailView } from './engagement_details';

export const EngagementRoute = () => {
  const { engagementService, categoryService } = useServiceProviders();
  const feedback = useFeedback();
  const analyticsContext = useAnalytics();
  const authContext = useSession();
  const { engagementFormConfig } = useEngagementFormConfig(engagementService);
  return (
    <EngagementProvider
      authContext={authContext}
      feedbackContext={feedback}
      analyticsContext={analyticsContext}
      categoryService={categoryService}
      engagementService={engagementService}
      engagementFormConfig={engagementFormConfig}
    >
      <ScrollToTop>
        <EngagementDetailView />
      </ScrollToTop>
    </EngagementProvider>
  );
};
