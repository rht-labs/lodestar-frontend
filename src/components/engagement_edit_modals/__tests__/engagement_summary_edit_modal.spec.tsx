import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { EngagementSummaryEditModal } from '../engagement_summary_edit_modal';
import { Engagement } from '../../../schemas/engagement';

describe('Point of Contact edit modal', () => {
  test('matches snapshot', () => {
    expect(
      render(
        <EngagementSummaryEditModal
          onSave={() => {}}
          formOptions={{}}
          isOpen={true}
          engagement={Engagement.fromFake(true)}
          onChange={() => {}}
        />
      )
    ).toMatchSnapshot();
  });

  test('When clicking the save button, the onSave method is called', async () => {
    const onSave = jest.fn();
    const { getByTestId } = render(
      <EngagementSummaryEditModal
        onSave={onSave}
        engagement={Engagement.fromFake(true)}
        formOptions={{}}
        isOpen={true}
        onChange={() => {}}
      />
    );
    await fireEvent.click(getByTestId('engagement-summary-save'));
    expect(onSave).toHaveBeenCalled();
  });
});
