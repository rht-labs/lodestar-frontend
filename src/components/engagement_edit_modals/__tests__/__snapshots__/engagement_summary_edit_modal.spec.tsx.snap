// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Point of Contact edit modal matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="pf-c-backdrop__open"
  >
    <div
      aria-hidden="true"
    />
    <div>
      <div
        class="pf-c-backdrop"
      >
        <div
          class="pf-l-bullseye"
        >
          <div
            aria-describedby="pf-modal-part-2"
            aria-labelledby="pf-modal-part-1"
            aria-modal="true"
            class="pf-c-modal-box pf-m-sm"
            id="pf-modal-part-0"
            role="dialog"
          >
            <button
              aria-label="Close"
              class="pf-c-button pf-m-plain"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 352 512"
                width="1em"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                  transform=""
                />
              </svg>
            </button>
            <header
              class="pf-c-modal-box__header"
            >
              <h1
                class="pf-c-modal-box__title"
                id="pf-modal-part-1"
              >
                Engagement Summary
              </h1>
            </header>
            <div
              class="pf-c-modal-box__body"
              id="pf-modal-part-2"
            >
              <section
                class="pf-c-page__main-section"
              >
                <form
                  class="pf-c-form"
                  novalidate=""
                >
                  <div
                    class="pf-c-form__group"
                  >
                    <div
                      class="pf-c-form__group-label"
                    >
                      <label
                        class="pf-c-form__label"
                        for="description"
                      >
                        <span
                          class="pf-c-form__label-text"
                        >
                          Description
                        </span>
                      </label>
                       
                    </div>
                    <div
                      class="pf-c-form__group-control"
                    >
                      <textarea
                        aria-invalid="false"
                        aria-label="engagement description"
                        class="pf-c-form-control pf-m-resize-vertical"
                        data-cy="description_field"
                        disabled=""
                        id="description"
                        name="description"
                        placeholder="Description and notes for the Engagement"
                      >
                        It's rocket science
                      </textarea>
                      
                    </div>
                  </div>
                  <div
                    class="pf-c-form__group"
                  >
                    <div
                      class="pf-c-form__group-label"
                    >
                      <label
                        class="pf-c-form__label"
                        for="engagement-location"
                      >
                        <span
                          class="pf-c-form__label-text"
                        >
                          Location
                        </span>
                      </label>
                       
                    </div>
                    <div
                      class="pf-c-form__group-control"
                    >
                      <input
                        aria-invalid="false"
                        class="pf-c-form-control"
                        data-cy="location_field"
                        data-testid="location-field"
                        disabled=""
                        id="location"
                        name="location"
                        placeholder="e.g. Pasadena, CA"
                        type="text"
                        value="Nashville, TN"
                      />
                      <div
                        aria-live="polite"
                        class="pf-c-form__helper-text"
                        id="engagement-location-helper"
                      >
                        Where will this be held?
                      </div>
                    </div>
                  </div>
                  <div
                    class="pf-c-form__group"
                  >
                    <div
                      class="pf-c-form__group-label"
                    >
                      <label
                        class="pf-c-form__label"
                        for="engagement-dates"
                      >
                        <span
                          class="pf-c-form__label-text"
                        >
                          Start and End Dates
                        </span>
                        <span
                          aria-hidden="true"
                          class="pf-c-form__label-required"
                        >
                           
                          *
                        </span>
                      </label>
                       
                    </div>
                    <div
                      class="pf-c-form__group-control"
                    >
                      <div
                        class="pf-c-input-group"
                        label="Engagement Duration"
                      >
                        <label
                          class="pf-c-input-group__text"
                          for="engagement-duration"
                        >
                          <svg
                            aria-hidden="true"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 448 512"
                            width="1em"
                          >
                            <path
                              d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm320-196c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM192 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM64 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
                              transform=""
                            />
                          </svg>
                        </label>
                        <input
                          aria-invalid="false"
                          aria-label="The start date."
                          class="pf-c-form-control"
                          data-cy="start_date_input"
                          disabled=""
                          id="start_date"
                          name="start_date"
                          type="date"
                          value="2020-06-01"
                        />
                        <input
                          aria-invalid="false"
                          aria-label="The end date"
                          class="pf-c-form-control"
                          data-cy="end_date_input"
                          disabled=""
                          id="end_date"
                          min="2020-08-03"
                          name="end_date"
                          type="date"
                          value="2020-07-01"
                        />
                      </div>
                      <div
                        aria-live="polite"
                        class="pf-c-form__helper-text"
                        id="engagement-dates-helper"
                      >
                        What is the duration?
                      </div>
                    </div>
                  </div>
                  <div
                    class="pf-c-form__group"
                  >
                    <div
                      class="pf-c-form__group-label"
                    >
                      <label
                        class="pf-c-form__label"
                        for="retirement"
                      >
                        <span
                          class="pf-c-form__label-text"
                        >
                          <span
                            title="Modifying the retirement date will impact the total cost of the residency.
Any changes to this field should be coordinated with Labs leadership."
                          >
                            Environment Retirement Date 
                            <svg
                              aria-hidden="true"
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style="vertical-align: -0.125em;"
                              viewBox="0 0 512 512"
                              width="1em"
                            >
                              <path
                                d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                                transform=""
                              />
                            </svg>
                          </span>
                        </span>
                      </label>
                       
                    </div>
                    <div
                      class="pf-c-form__group-control"
                    >
                      <div
                        class="pf-c-input-group"
                        label="Retirement Date"
                      >
                        <input
                          aria-invalid="false"
                          aria-label="Environment Retirement Date"
                          class="pf-c-form-control"
                          data-cy="retirement_date_input"
                          disabled=""
                          id="archive_date"
                          min="2020-07-01"
                          name="archive_date"
                          type="date"
                          value="2020-07-30"
                        />
                      </div>
                      
                    </div>
                  </div>
                </form>
              </section>
              <section
                class="pf-c-page__main-section"
              >
                <div
                  style="margin-top: 1rem;"
                >
                  <div
                    class="pf-l-flex pf-m-justify-content-flex-end"
                  >
                    <div
                      class=""
                    >
                      <div>
                        <button
                          class="pf-c-button pf-m-primary"
                          data-cy="save_summary_card"
                          data-ouia-component-type="PF4/Button"
                          data-ouia-safe="true"
                          data-testid="engagement-summary-save"
                          type="button"
                        >
                          Save
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </section>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
